submariner/OVN-Interconnect.md

```diff
@@ -0,0 +1,126 @@
+will remain as it is today.
+
+* If a zone configuration is detected
+
+1) In the gateway node zone a new routing rule can be added to forward the traffic destined for Submariner remote CIDRs
+to the Submariner router. This is required for the traffic coming from pods in a node that are in a different zone.
+
+2) In the non-gateway zones a route will be added for the traffic destined for remote cluster CIDR to forward it to the gateway node
+transit switch.
+
+#### Pros of Option1
+
+This makes the least changes than option 2 and can be delivered within the time frame we have.
+
+#### Cons of Option1
+
+It depends on OVN labels/annotations for programming the routes, which means a change in these would break Submariner. Also given the plans
+to change the OVN -Submariner architecture this has to be reworked in future.
+
+### Option2: Submariner OVN Re-architecture
+
+There is an effort to optimize Submariner OVN integration. As a part of this Submariner Router and related ports and join switch shall be
+removed.  The OVN gateway router needs to be configured not to SNAT the traffic destined to local Cluter CIDR/ Global net CIDRS(This is now
+the default behaviour of OVN, we need it to test it though).
+
+Further, with this optimization Submariner should stop configuring the OVN cluster router with Logical Route policies using OVN APIs
+directly. OVN should expose CRD, which Submariner can use to program the OVN cluster router with the required Logical router policy and
+routes. If done, the Submariner need not be aware of the OVN infrastructure details like zones and transit switches. An OVN plugin (or a
+component in OVN) can consume the CRD and program the routing rules and provision the infrastructure required by Submariner.
+
+While this is a cleaner and maintainable implementation it will not be possible to deliver it by the next Submariner release. Also, we need
+to agree upon the APIs required in OVN and their implementation.
+
+#### Pros of Option2
+
+This will lead to a more maintainable code and fewer moving parts and this is how we plan Submariner-OVN integration will be in future.
+
+#### Cons of Option2
+
+It takes more time to implement this option. Submariner changes need to wait until OVN changes required for Submariner are done.
+
+## Design Details
+
+### Backward Compatibility
+
+<!-- Any backward compatibility concerns -->
+
+### Alternatives
+
+#### Route Advertisement
+
+Another alternative was explored which could leverage the [route advertisement](
+https://docs.ovn.org/en/latest/tutorials/ovn-interconnection.html#route-advertisement) feature in OVN. The idea was to use this capability
+to advertise Submariner routes. But this feature seems to be not available in the OVN Kubernetes IC implementation.
+
+## External Dependencies
+
+<!-- Any external dependencies this proposal may have -->
+
+## User Impact
+
+<!-- Optional. Any impact on users this change may have. -->
```

Comment:
Ideally, if done right, there should not be any user impact here. But we need to ensure that this is really the case. 

I think we need to explicitly cover:

1. Greenfield/new clusters deployed with OVN IC and Submariner 
2. Brownfield/upgrade scenario from existing OVN to OVN IC - here I am assuming there is going to be some downtime during upgrade?

Reply:
I thought of adding the section once Option1/Option2 is finalized.